server/queues/tasks/ImportTask.ts

```diff
@@ -562,4 +545,25 @@ export default abstract class ImportTask extends BaseTask<Props> {
       attempts: 1,
     };
   }
+
+  private async preprocessDocUrlIds(data: StructuredImportData) {
+    for (const doc of data.documents) {
+      // check DB only if urlId is present in the input.
+      if (doc.urlId) {
+        const existing = await Document.unscoped().findOne({
+          attributes: ["id"],
+          paranoid: false,
+          where: {
+            urlId: doc.urlId,
+          },
+        });
+
+        if (!existing) {
+          continue;
+        }
+      }
+
+      doc.urlId = generateDocUrlId();
```

Comment:
The chances of a collision with `{slugifiedTitle}-{urlId}` should be relatively low within a single machine.
To make it more resilient, retry can be added here; but I think it might be an overkill.

Reply:
Looks like the chances of collision are 1 in ~70millon, I can live with that

https://www.wolframalpha.com/input?i=1%2F2%2Bsqrt%281%2F4-2*%2836%5E10%29*ln%280.5%29%29

Reply:
_nit_:  The collision chances are 1 in ~1 billion, I believe (62 characters instead of 36).
https://www.wolframalpha.com/input?i=1%2F2+%2B+sqrt%281%2F4+%2B+2*%2862%5E10%29*ln%282%29%29

The actual collisions that we're worried about (_{title}-{urlId}_) will be even lower than `urlId` on its own.